tests/components/zha/test_init.py

```diff
@@ -70,3 +71,27 @@ async def test_migration_from_v1_wrong_baudrate(hass, config_entry_v1):
     assert CONF_USB_PATH not in config_entry_v1.data
     assert CONF_BAUDRATE not in config_entry_v1.data[CONF_DEVICE]
     assert config_entry_v1.version == 2
+
+
+@pytest.mark.skipif(
+    MAJOR_VERSION != 0 or (MAJOR_VERSION == 0 and MINOR_VERSION >= 112),
+    reason="Not applicaable for this version",
+)
+@pytest.mark.parametrize(
+    "zha_config",
+    (
+        {},
+        {CONF_USB_PATH: "str"},
+        {CONF_RADIO_TYPE: "ezsp"},
+        {CONF_RADIO_TYPE: "ezsp", CONF_USB_PATH: "str"},
+    ),
+)
+async def test_config_depreciation(hass, zha_config):
+    """Test config option depreciation."""
+    await async_setup_component(hass, "persistent_notification", {})
+
+    with patch(
+        "homeassistant.components.zha.async_setup", return_value=True
+    ) as setup_mock:
+        assert await async_setup_component(hass, DOMAIN, {DOMAIN: zha_config})
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)